### ISSUE TITLE
Project rename input loses focus when an auto-save is triggered, causing you to type into the document.

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#1306]
Steps to reproduce
- Make a change to one of the files in the project
- Immediately after, star renaming your project
- While you're changing the name, the auto-save kicks in
- Editor regains focus, and you are typing into the edited file now

![rename-focus](https://cloud.githubusercontent.com/assets/25212/11818787/2cd9c89a-a311-11e5-9ee4-0d2bad3565aa.gif)

**Comment by humphd** [comment]
We dug into this in class a bit, and if someone wants to try and fix it, the issue is here:

https://github.com/mozilla/brackets/blob/master/src/document/DocumentCommandHandlers.js#L864

After a save happens in Brackets, focus is reset to the editor when the promise finishes.  You could fix this a few ways:

* Just not do that :)  I am unclear why we need to refocus after save, but the pattern is repeated throughout the code, so it might wise to keep it.
* Add a flag that can be toggled from the Bramble API to not steal focus while we are waiting on the user to finish with it in Thimble.  You'd have to add a new set of `bramble.*()` API calls to enable/disable focus while the user is typing in an input box in Thimble.

**Comment by omytryniuk** [comment]
@humphd, I think this bug is exactly what I would like to work on.

**Comment by omytryniuk** [comment]
I have reproduced the issue and spent some time trying to understand what can be wrong. Seems like [this line](https://github.com/mozilla/brackets/blob/master/src/document/DocumentCommandHandlers.js#L879) of code causes the problem.
What I see, when we work with an editor, the code is saved every 5-10 seconds, and after that .focusActivePane() function is called. When we focus on a title, the app still counts the time (5-10 seconds) and calls .focusActivePane() function again(that focus on the editor). I will spend more time on it, and hope to have some progress.

**Comment by humphd** [comment]
Great.  Why don't you try an experiment to remove that auto-focus-on-save code (just comment it out) and see how the app responds.

**Comment by omytryniuk** [comment]
@humphd It is exactly what I did  yesterday and it solved the problem. What I meant, I want to spend more time just trying to see side-affects of such changes.

**Comment by Pomax** [comment]
This issue seems to have stalled a little, @omytryniuk did you find any side effects? If not, did you have a chance to form a PR?

**Comment by omytryniuk** [comment]
Good day Mr. Kamermans. I did not find any side effects, because the part of the code(result.always) is only executed when we type inside the editor. I have removed the function. At the same time, I did not remove focusActivePane() in MainViewManager, because it is used by other services. Here is the [PR](https://github.com/mozilla/brackets/pull/669). It is Brackets bug, that is why I've formed the PR for Brackets. Thank you!
